submissions/andysmokk/classic-frogger-game/app.js

```diff
@@ -0,0 +1,117 @@
+// Enemies our player must avoid
+const INITIAL_POSITION_PLAYER_X = 203;
+const INITIAL_POSITION_ENEMIES_X = -200;
+const INITIAL_POSITION_ENEMY_ONE_Y = 62;
+const INITIAL_POSITION_ENEMY_TWO_Y = 145;
+const INITIAL_POSITION_ENEMY_THREE_Y = 228;
+const INITIAL_POSITION_PLAYER_Y = 390;
+const BLOCK_HEIGHT = 83;
+const BLOCK_WIDTH = 101;
+const MIN_SPEED = 100;
+const MAX_SPEED = 400;
+
+var Enemy = function (x, y) {
+  this.x = x;
+  this.y = y;
+  this.speed = this.randomInteger();
+  this.sprite = "images/enemy-bug.png";
+};
+
+Enemy.prototype.update = function (dt) {
+  this.x += this.speed * dt;
+
+  if (this.x > 500) {
+    this.x = INITIAL_POSITION_ENEMIES_X;
+    this.speed = this.randomInteger();
+  }
+
+  if (
+    player.y < this.y + 50 &&
+    player.y + 50 > this.y &&
+    player.x < this.x + 75 &&
+    player.x + 75 > this.x
```

Comment:
Please find a requirement among those listed/linked by bot, post it here as a comment and also fix code accordingly.

Reply:
Classes do not refer to any global variables, like global variable player, which is an instance of Player class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with.